runtime/compiler/ilgen/Walker.cpp

```diff
@@ -6754,6 +6754,23 @@ TR_J9ByteCodeIlGenerator::genNewArray(int32_t typeIndex)
         }
      }
 
+   // Special case for handling Arrays.copyOfRangeByte when the caller of caller is String<init>(AbstractStringBuilder, Void).
+   // The caller is always Arrays.copyOfRange since Arrays.copyOfRangeByte is a private method.
+   if ((comp()->getInlineDepth() > 1) && !comp()->isPeekingMethod() && !generateArraylets
+       && _methodSymbol->getRecognizedMethod() == TR::java_util_Arrays_copyOfRangeByte)
+      {
+      int32_t callerIndex = comp()->getCurrentInlinedCallSite()->_byteCodeInfo.getCallerIndex();
+      int32_t callerOfCallerIndex = comp()->getInlinedCallSite(callerIndex)._byteCodeInfo.getCallerIndex();
```

Comment:
I am skip checking the first caller since I expect it to always be `Arrays.copyOfRange` since `copyOfRangeByte` is private. I also check the second caller signature so it must be safe even though non of these method are intrinsic.

Reply:
I understand `copyOfRangeByte` is private and there might be only one caller of that method, that does not mean we can take this assumption. Some other method from Arrays class can call it, so as can be done through reflection. I do not think we can take such assumptions.